cpu/sam0_common/periph/adc.c

```diff
@@ -102,8 +102,8 @@ static int _adc_configure(adc_res_t res)
     /* Power On */
     PM->APBCMASK.reg |= PM_APBCMASK_ADC;
     /* GCLK Setup */
-    GCLK->CLKCTRL.reg = (uint32_t)(GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0 |
-                        (GCLK_CLKCTRL_ID(ADC_GCLK_ID)));
+    GCLK->CLKCTRL =
+        (GCLK_CLKCTRL_Type){{.CLKEN=true, .GEN=adc_config.gclk_src, .ID=(uint16_t)GCLK_CLKCTRL_ID_ADC_Val}};
```

Comment:
Why do no use CLKCTRL.reg and a mask ?
@aabadie is it fine with the RIOT's coding style ?

Reply:
Maybe some spaces are missing around `=` and after/before curly braces. In general, this is not very nice in term of readability but I think this is very subjective.

Reply:
I went with this style, because I thought there was a preference to move towards bit addressed fields, rather than masks and logical operations. I was thinking that in a separate PR I would clean up the rest of the ADC driver. But maybe I have misread the trend there?

Personally, I prefer the bitfields. I got tired of fixing bugs in our own firmware code related to incorrectly applied shifts and other field mask operations. Going bit direct has eliminated a whole class of errors.

That said, when in Rome... do as the Romans. Format/style wars generate a lot of heat and only a little light. So you guys say what style you want, and I'll do that.

Reply:
TBH, It's not my war. But I do believe we have some code zealots among the maintainers ;)
I'll test your PR on real hardware this weekend just in case

Reply:
two things:
- bitfields are fine, as long as they don't give any computational overhead. So a quick test building this and building it once more using logical ops should give an idea
- if the notation above is kept, I strongly suggest to format it in a readable way, e.g. something like:
```c
GCLK->CLKCTRL = (GCLK_CLKCTRL_Type) {{
    .CLKEN = true,
    .GEN = adc_config.gclk_src, 
    .ID = (uint16_t)GCLK_CLKCTRL_ID_ADC_Val
}};
```